system/jlib/jsocket.cpp

```diff
@@ -5081,51 +5097,97 @@ class CSocketEpollThread: public CSocketBaseThread
 
 class CSocketEpollHandler: implements ISocketSelectHandler, public CInterface
 {
-    CSocketEpollThread *epollthread;
+    CIArrayOf<CSocketEpollThread> threads;
     CriticalSection sect;
+    bool started;
     StringAttr epolltrace;
+    unsigned hdlPerThrd;
 public:
     IMPLEMENT_IINTERFACE;
-    CSocketEpollHandler(const char *trc)
-        : epolltrace(trc)
+    CSocketEpollHandler(const char *trc, unsigned _hdlPerThrd=(unsigned)-1)
+        : started(false), epolltrace(trc), hdlPerThrd(_hdlPerThrd)
     {
-        epollthread = new CSocketEpollThread(epolltrace);
+        if (hdlPerThrd == 0)
+            hdlPerThrd = epoll_hdlPerThrd;
```

Comment:
minor: as prev. comment, think it would clearer without the default param on the constructor here and this check, this class in internal and the caller validates the param value already.